embedded/src/main/java/org/wildfly/core/embedded/EmbeddedServerFactory.java

```diff
@@ -293,8 +296,14 @@ private static ModuleLoader setupModuleLoader(String modulePath, String... syste
             }
             WildFlySecurityManager.setPropertyPrivileged("jboss.modules.system.pkgs", packages.toString());
 
-            // Get the module loader
-            return Module.getBootModuleLoader();
+            // setup the module class so we can reload it and reset the static properties later if we need to
+            URL[] urls = new URL[] {new File(modulePath).toURI().toURL(), new File(modulePath + "/../jboss-modules.jar").toURI().toURL()};
+            ClassLoader classloader = new URLClassLoader(urls, ClassLoader.getSystemClassLoader());
+
+            Class clazz = classloader.loadClass("org.jboss.modules.Module");
+            return (ModuleLoader)clazz.getMethod("getBootModuleLoader").invoke(clazz);
```

Comment:
I don't get how it works though. :) If the Module class is visible in the system classloader, URLClassLoader should be returning that as it's a parent first CL. And if its not visible and Module is loaded from urls[0], then how can the result of getBootModuleLoader be cast to the correct ModuleLoader?